### PR TITLE
CommonServerPWSH - Integration context - Fix demisto server version compatability

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32274

## Description
Handling get intergation context, If demisto version is greater eqaul then 6.0.0, Integration context should be in context attribute otherwise in the same level.

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests

## TBD
 - [x] Waiting for customer eval.

### CODE REVIEW

**Comment by GalRabin** [comment]
@kirbles19
I will appriciate your review on:
- [ ] [Release note](https://github.com/demisto/content/pull/10789/files#diff-7aa25ff121cbf2d2b5fb62d8274d9a664b0e7948d042fbe118c95cf6c33764c3)

File: Packs/Base/Scripts/CommonServerPowerShell/script-CommonServerPowerShell.yml [hunk 1]
```diff
@@ -0,0 +1,735 @@
+commonfields:
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPowerShell/script-CommonServerPowerShell.yml:1]: @GalRabin I think you added this file by mistake

File: Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1 [hunk 2]
```diff
@@ -268,7 +268,14 @@ class DemistoObject {
         if ( -not $this.IsIntegration ) {
             throw "Method not supported"
         }
-        return $this.ServerRequest(@{type = "executeCommand"; command = "getIntegrationContext"; args = @{ } }).context
+        $integration_context = $this.ServerRequest(@{type = "executeCommand"; command = "getIntegrationContext"; args = @{ } })
+        # When Demisto Version is greater equal then "6.0.0".  integration_context will be under "context" attribute.
+        $this.DemistoVersion().version -match "\d{1,2}\.\d{1,2}\.\d{1,2}" | Out-Null
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1:273]: Not sure why we are doing this. This logic should be merged into the VersionEqualGreaterThen.

**Comment by glicht** [comment]
Please check regarding: `SetIntegrationContext`. I think the versioning format changed between 6.0 and 6.1.

Also, let's create a test playbook that verifies set/get integration context.

**Comment by GalRabin** [comment]
> Please check regarding: `SetIntegrationContext`. I think the versioning format changed between 6.0 and 6.1.
> 
> Also, let's create a test playbook that verifies set/get integration context.

Verified it:
image

File: Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1 (hunk 2 above)
Comment (GalRabin) [reply Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1:273]: Just for unit-test reason, you right I will mock it and insert it in to VersionEqualGreaterThen

File: Packs/Base/Scripts/CommonServerPowerShell/script-CommonServerPowerShell.yml (hunk 1 above)
Comment (GalRabin) [reply Packs/Base/Scripts/CommonServerPowerShell/script-CommonServerPowerShell.yml:1]: Right

Comment (GalRabin) [reply Packs/Base/Scripts/CommonServerPowerShell/script-CommonServerPowerShell.yml:1]: Right sorry for that.

**Comment by GalRabin** [comment]
Updating, Shared  it with the customer and it solved the issue.

File: Packs/Base/ReleaseNotes/1_6_4.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPowerShell
+- GetIntegrationContext method fix in DemistoObject, Demisto version 5.5.0 and above will be supported.
```
Comment (yaakovi) [inline Packs/Base/ReleaseNotes/1_6_4.md:4]:
```suggestion
- GetIntegrationContext method fix in DemistoObject, Cortex XSOAR version 5.5.0 and above will be supported.
```

**Review by glicht — CHANGES_REQUESTED** [review]
Please create a test playbook that verifies set/get integration context.

**Comment by GalRabin** [comment]
> Please create a test playbook that verifies set/get integration context.
@glicht 
The test playbook should be related to the integration or it should be stand alone test with a dummy integration which only valiadte set/get integration context?

**Comment by GalRabin** [comment]
Updating, Validated support on self service instances 5.5.0 / 6.0.0 demisto versions.
Also validated with the customer.

File: Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1 [hunk 4]
```diff
@@ -603,4 +609,13 @@ function FileResult([string]$file_name, [string]$data, [string]$file_type) {
         "File" = $file_name
         "FileID" = $temp
     }
+}
+
+function DemistoVersionEqualGreaterThen([string]$version) {
```
Comment (yaakovi) [inline Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1:614]: this is the common name
```suggestion
function DemistoVersionGreaterEqualThen([string]$version) {
```

**Review by yaakovi — APPROVED** [review]

**Comment by GalRabin** [comment]
@glicht 
As you requested, I added the following:
1. Integration with set/get interation context command (Powershell).
2. Test-playbook which test this functionallity from version 5.5.0.

I will skip docs review as you suggested.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
+This integration allows interacting with XSOAR server, mostly for internal use.
```
Comment (glicht) [inline Packs/DeveloperTools/Integrations/XsoarServerPowershell/README.md:1]: @GalRabin let's remove this file. No need for documentation in the site for this integration.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+category: Messaging
+commonfields:
+  id: XsoarServerPowershell
+  version: -1
+configuration: []
+description: This integration allows interacting with XSOAR server.
+display: XSOAR Server Powershell
```
Comment (glicht) [inline Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml:7]: @GalRabin let's change the display name:
```suggestion
display: XSOAR Powershell Testing
```

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
+category: Messaging
+commonfields:
+  id: XsoarServerPowershell
+  version: -1
+configuration: []
+description: This integration allows interacting with XSOAR server.
```
Comment (glicht) [inline Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml:6]: Change the description
```suggestion
description: Powershell Integration for testing that Powershell support is working as expected.
```

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml (hunk 6 above)
Comment (glicht) [reply Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml:7]: You will need to also update the display name in the release notes and detailed description.

**Review by glicht — COMMENTED** [review]
It looks good. See my comments about naming and descrption.

Also please change the command prefix to: `pwsh-test`. So it is clear that the command is for testing.

**Comment by glicht** [comment]
@GalRabin you can also mark this integration as `mock disabled` as it doesn't do network traffic.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.ps1 [hunk 8]
```diff
@@ -0,0 +1,90 @@
+. $PSScriptRoot\CommonServerPowerShell.ps1
+
+$script:INTEGRATION_NAME = "XSOAR Server Powershell"
+$script:COMMAND_PREFIX = "xsoar"
+$script:INTEGRATION_ENTRY_CONTEX = "XSOAR"
+
+#### HELPER FUNCTIONS ####
+
+function SetIntegrationContextCommand() {
```
Comment (glicht) [inline Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.ps1:9]: It looks like you defined this function twice. See line 34.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/README.md (hunk 5 above)
Comment (GalRabin) [reply Packs/DeveloperTools/Integrations/XsoarServerPowershell/README.md:1]: Done

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml (hunk 6 above)
Comment (GalRabin) [reply Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml:7]: Changed in all required places.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml (hunk 7 above)
Comment (GalRabin) [reply Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.yml:6]: Change in release note, descriptions.

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.ps1 (hunk 8 above)
Comment (GalRabin) [reply Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarServerPowershell.ps1:9]: Its a mistake, It should be `TestModuleCommand` changed.

**Comment by GalRabin** [comment]
> It looks good. See my comments about naming and descrption.
> 
> Also please change the command prefix to: `pwsh-test`. So it is clear that the command is for testing.

Changed

**Comment by GalRabin** [comment]
> @GalRabin you can also mark this integration as `mock disabled` as it doesn't do network traffic.

Disabled

**Review by yaakovi — APPROVED** [review]
please replace the integration image to be power shell related

**Review by glicht — COMMENTED** [review]
Please remove the following empty file: XsoarPowershellTesting.Tests.ps1

File: Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarPowershellTesting.ps1 [hunk 9]
```diff
@@ -0,0 +1,90 @@
+. $PSScriptRoot\CommonServerPowerShell.ps1
+
+$script:INTEGRATION_NAME = "Xsoar Powershell Testing"
```
Comment (glicht) [inline Packs/DeveloperTools/Integrations/XsoarServerPowershell/XsoarPowershellTesting.ps1:3]:
```suggestion
$script:INTEGRATION_NAME = "XSOAR Powershell Testing"
```

**Review by glicht — APPROVED** [review]
Looks good

**Pull request merged**